platform/Sqlite.roc

```diff
@@ -0,0 +1,710 @@
+
+## Decode a [Value] to a [Nullable U64].
+nullable_u64 : Str -> SqlDecode (Nullable U64) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_u64 = nullable_int_decoder Num.toU64Checked
+
+## Decode a [Value] to a [Nullable U32].
+nullable_u32 : Str -> SqlDecode (Nullable U32) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_u32 = nullable_int_decoder Num.toU32Checked
+
+## Decode a [Value] to a [Nullable U16].
+nullable_u16 : Str -> SqlDecode (Nullable U16) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_u16 = nullable_int_decoder Num.toU16Checked
+
+## Decode a [Value] to a [Nullable U8].
+nullable_u8 : Str -> SqlDecode (Nullable U8) [FailedToDecodeInteger [OutOfBounds]]UnexpectedTypeErr
+nullable_u8 = nullable_int_decoder Num.toU8Checked
+
+## Decode a [Value] to a [Nullable F64].
+nullable_f64 : Str -> SqlDecode (Nullable F64) [FailedToDecodeReal []]UnexpectedTypeErr
+nullable_f64 = nullable_real_decoder Ok
+
+## Decode a [Value] to a [Nullable F32].
+nullable_f32 : Str -> SqlDecode (Nullable F32) [FailedToDecodeReal []]UnexpectedTypeErr
+nullable_f32 = nullable_real_decoder (\x -> Num.toF32 x |> Ok)
+
+# TODO: Mising Num.toDec and Num.toDecChecked
+# nullable_dec = nullable_real_decoder Ok
+
+# internal use only
+internal_to_external_error : InternalSqlite.SqliteError -> Error
+internal_to_external_error = \{ code, message } ->
+    SqlError (code_from_i64 code) message
+
+# internal use only
+code_from_i64 : I64 -> InternalSqlite.SqliteErrCode
+code_from_i64 = \code ->
+    if code == 1 || code == 0 then
+        ERROR
+    else if code == 2 then
+        INTERNAL
+    else if code == 3 then
+        PERM
+    else if code == 4 then
+        ABORT
+    else if code == 5 then
+        BUSY
+    else if code == 6 then
+        LOCKED
+    else if code == 7 then
+        NOMEM
+    else if code == 8 then
+        READONLY
+    else if code == 9 then
+        INTERRUPT
+    else if code == 10 then
+        IOERR
+    else if code == 11 then
+        CORRUPT
+    else if code == 12 then
+        NOTFOUND
+    else if code == 13 then
+        FULL
+    else if code == 14 then
+        CANTOPEN
+    else if code == 15 then
+        PROTOCOL
+    else if code == 16 then
+        EMPTY
+    else if code == 17 then
+        SCHEMA
+    else if code == 18 then
+        TOOBIG
+    else if code == 19 then
+        CONSTRAINT
+    else if code == 20 then
+        MISMATCH
+    else if code == 21 then
+        MISUSE
+    else if code == 22 then
+        NOLFS
+    else if code == 23 then
+        AUTH
+    else if code == 24 then
+        FORMAT
+    else if code == 25 then
+        RANGE
+    else if code == 26 then
+        NOTADB
+    else if code == 27 then
+        NOTICE
+    else if code == 28 then
+        WARNING
+    else if code == 100 then
+        ROW
+    else if code == 101 then
+        DONE
+    else
+        crash "unsupported Sqlite error code $(Num.toStr code)"
```

Comment:
While I'm thinking about this. We probably shouldn't crash here. We should instead make a generic error that wraps the code. Some sort of `Unknown` error.